### DISCUSSION TITLE
Can ASDF validation handle newer JSON schema versions as ASDF schemas?

### DISCUSSION THREAD

**Discussion by ketozhang** [spacetelescope/pyasdf#1793] [Q&A] [answered]
ASDF schemas are basically JSON schemas, but converted to YAML...

> These schemas, though expressed in YAML, are structured according to the [JSON Schema Draft 4](https://json-schema.org/specification-links.html#draft-4) specification.
> — https://asdf.readthedocs.io/en/latest/asdf/extending/schemas.html

I am interested [JSON Schema Draft 2020-12](https://json-schema.org/draft/2020-12/release-notes.html) largely because [Pydantic can generate them](https://docs.pydantic.dev/latest/concepts/json_schema/#generating-json-schema).

**Comment by braingram** [comment, marked as answer]
The current ASDF schemas mostly use the [http://stsci.edu/schemas/yaml-schema/draft-01](https://github.com/asdf-format/asdf-standard/blob/main/resources/schemas/stsci.edu/yaml-schema/draft-01.yaml) metaschema which is largely https://json-schema.org/draft-04/schema with some additions for YAML and ASDF features.

There is not currently support for other jsonschema drafts. It looks like pydantic allows you to customize the [schema generation](https://docs.pydantic.dev/latest/concepts/json_schema/#customizing-the-json-schema-generation-process) (to allow you to generate schemas of other dialects). Have you tried this for generating either draft-04 or ASDF schemas?

**Reply by ketozhang** [reply]
Yes, I've started experimenting with this. The task is given a jsonschema 2020-12 schema, coerce it into the one you want. It's not a trivial task, and I need handle all the differences.

Would `asdf.schema.check_schema()` alone be enough to validate (i.e., check against `$schema` field)? I have yet failed to pass this check.

**Reply by braingram** [reply]
`asdf.schema.check_schema` is a start but given the allowable options for the schemas and how they relate to asdf (through the converters and tags) it's not sufficient to fully evaluate a schema. One reason is that the degree of schema flexibility is an option to the user (as is the case for the base jsonschema).

Would you share an example of the draft 2020-12 schema that you're hoping to convert? Also are you able to provide some more details about what you're hoping to do? Seeing the code would be ideal.
